shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizReducer.kt

```diff
@@ -228,6 +230,19 @@ class StepQuizReducer(
                         ProblemsLimitReachedModalHiddenHyperskillAnalyticEvent(stepRoute.analyticRoute)
                     )
                 )
+            is Message.ParsonsProblemOnboardingModalShownEventMessage ->
+                state to setOf(
+                    Action.LogAnalyticEvent(
+                        ParsonsProblemOnboardingModalShownHyperskillAnalyticEvent(stepRoute.analyticRoute)
+                    ),
+                    Action.SaveParsonsProblemOnboardingModalShownCacheFlag
+                )
```

Comment:
I suggest naming all the messages without the `Event` postfix to not expect some concrete behavior reading the name of the event.
In this case, `ParsonsProblemOnboardingModalShownEventMessage` suppose not only analytics sending.

Reply:
I agree, very good point :+1: